aws-api/src/main/java/com/amplifyframework/api/aws/AWSApiPlugin.java

```diff
@@ -361,65 +363,61 @@ private boolean isReadRestrictingOwner(AuthRule authRule) {
 
     private boolean isReadRestrictingStaticGroup(AuthRule authRule) {
         return AuthStrategy.GROUPS.equals(authRule.getAuthStrategy())
-            && authRule.getGroups() != null && !authRule.getGroups().isEmpty()
+            && !authRule.getGroups().isEmpty()
```

Comment:
Why remove the null check?

Reply:
It's redundant because `AuthRule#getGroups()` explicitly has `@NonNull` return value :)